### DISCUSSION TITLE
'연습모드 결과 모달' 구현했습니다.

### DISCUSSION THREAD

**Discussion by sumi-0011** [msdio/Tamago#239] [Show and tell]
## 구현 사진

<img width="1110" alt="스크린샷 2023-03-31 오후 4 00 16" src="https://user-images.githubusercontent.com/49177223/229046888-2fcc9ccb-3d74-4da6-80c5-03c4ae50eabd.png">

## 필요한 Props
```
export interface TypingResultType {
  typingTime: number;
  typingAccuracy: number;
  typingWpm: number;
  typingSpeed: number;
}

interface ResultModalProps {
  isOpen: boolean;
  result: TypingResultType;
  onClose: () => void;
  endTime: Date; // 게임 종료 시간

  title?: string; // 긴 글의 경우 title이 존재
}
```

## 참고
- 하트 모양이 이상한것은 bookmark icon을 변경하면 자연스럽게 해결 될것 같습니다. 
- 북마크 기능은 아직 준비중이므로, 준비중이라는 alret을 띄우도록 구현했습니다. 
- `이어하기`보다는 타이핑 게임을 처음부터 다시시작해야하므로 `다시하기`가 맞다고 생각해 바꾸었는데 괜찮을까요? (나가기 모달에서 그만하기를 누른순간에 게임을 끝내는게 맞다고 생각합니다. )

---
의견 남겨주세요! 빠르게 반영하겠습니다.

**Comment by sumi-0011** [comment]
@coolsmart2 확인하고 긴글에도 적용해주세요!

**Reply by coolsmart2** [reply]
긴 글은 이어하기가 짧은 글이랑 달라 컴포넌트 파라미터를 수정해서 사용할게요

**Comment by msdio** [comment]
좋아요!

props 타입 중 `TypingResultType`은 긴 글과 짧은 글 모두에서 사용된다면, types 폴더로 빼는게 어떨까요?

> 이어하기보다는 타이핑 게임을 처음부터 다시시작해야하므로 다시하기가 맞다고 생각해 바꾸었는데 괜찮을까요? (나가기 모달에서 그만하기를 누른순간에 게임을 끝내는게 맞다고 생각합니다. )

실전인 경우에는 다시하기, 연습인 경우에는 이어하기가 맞는 것 같아요!

**Reply by sumi-0011** [reply]
`TypingResultType`는 types폴더로 분리하겠습니다.
